pages/learn/more/anti-patterns.md

```diff
@@ -0,0 +1,96 @@
+
+### Running docker-in-docker with host networking and iptables enabled
+(can be folded into the previous issue, leaving for completeness at review)
+
+### Manually changing the system clock, or blocking NTP requests
+Since {{ $names.os.lower }} & the supervisor communicate with {{ $names.cloud.lower }} using an HTTP API, it is important that time is synchronized on the device to ensure that certificates are trusted. {{ $names.os.upper }} provides a [number of mechanisms][time-sync] to keep time as up-to-date as possible, but ensuring [NTP is accessible over the network][networking-reqs] is critical.
+
+## Local storage
+
+### Writing to files in the container file system (and not a volume)
+Commonly, users misunderstand the distinction between the container's file system and a volume. Named volumes are preserved across updates unless specifically dereferenced or destroyed, while anything written to the container's file system will be purged during the next application update. For an in-depth comparison of the two, it is recommended to complete the following [services masterclass][services-mc]. Additionally, the [supervisor provides an API][supervisor-api] to manage named volumes.
+
+### Running out of disk space
+One of the most common issues moving from a project to a product is considering what is written to local disk. Often, either logs or important data are written locally but without giving thought to the data's lifecycle. It is important to ensure that some mitigations are in place to prevent any local storage from filling. [Some examples include][logging-solutions]:
+* configuring logrotate
+* external logging service
+* logging only to stderr/stdout to take advantage of {{ $names.engine.lower }}'s built-in logging features
+
+### Excessive writes causing SD / Flash failure / corruption
+If the device type uses flash memory or an SD card, any write-heavy workload will cause that storage medium to wear prematurely and may cause devices to crash irrecoverably. Limiting any writing to temporary locations in-memory is one strategy to avoid having to replace SD cards in the field. There are further SD card recommendations [available here][sd-cards]
+
+## Host OS services
+
+### Udev rules in container that conflict with host
+needs more information
+
+### Modifying the OS image
+Any modifications made to the host OS before flashing or at runtime carry undue risk. If the host OS has been modified in some way (other than connection details), any subsequent behavior cannot be guaranteed.
+
+### Overwriting the D-Bus socket
+needs more information
```

Comment:
Indeed. I see there are few of these lines around.

Reply:
This list was cribbed from @shaunmulligan's notes, but in some cases I'm not 100% what he means or has seen before. If this one (or the others) seems obvious to you, please help me write it!